test/sequential/test-net-connect-emfile.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+var net = require('net');
+var common = require('../common');
+var assert = require('assert');
+
+console.log('Test to check the error message format duing EMFILE error');
+
+if (process.platform === 'win32') {
+  console.log('Skipping test, no RLIMIT_NOFILE on Windows.');
+  return;
+}
+
+// Server listens
+net.createServer().listen(common.PORT);
+
+// Connect to server
+for (var i = 0; i < 100000; i++) {
+  net.connect({
+    port: common.PORT
+  });
+}
+
+process.on('uncaughtException', function(er) {
+  assert.equal(er.message, 'connect EMFILE 127.0.0.1:' + common.PORT);
+  assert.equal(er.code, 'EMFILE');
+  assert.equal(er.syscall, 'connect');
+  assert.equal(er.port, common.PORT);
+  console.log('Test ok');
+  process.exit(0);
```

Comment:
IMO this is theoretically safer. If an exception happened while trying to close the server it would enter a loop. No idea how that could happen, but it's a theoretical possibility.